File: WindowsFormsApplication1/WindowsFormsApplication1/Prichal.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WindowsFormsApplication1
{
    public class Prichal
    {
        List<ClassArray<Transport>> prichalStages;
        int countPlaces = 20;
        int placeSizeWidth = 210;
        int placeSizeHeight = 80;
        int currentLevel;
        public int getCurrentLevel { get { return currentLevel; } }
        public Prichal(int countStages)
        {
            prichalStages = new List<ClassArray<Transport>>();
            ClassArray<Transport> prichal;
            for (int i = 1; i < countPlaces; i++)
            {
                prichal = new ClassArray<Transport>(countPlaces, null);
                prichalStages.Add(prichal);
            }


        }

        public void LevelUp()
        {
            if (currentLevel + 1 < prichalStages.Count)
            {
                currentLevel++;
            }
        }
        public void LevelDown()
        {
            if (currentLevel > 0)
            {
                currentLevel--;
            }
        }
        public int PutBoatPrichal(Transport boat)
        {
            return prichalStages[currentLevel] + boat;
        }
        public Transport GetBoatPrichal(int ticket)
        {
            return prichalStages[currentLevel] - ticket;
        }

        public void Draw(Graphics g)
        {
            DrawMarking(g);
            for (int i = 0; i < countPlaces; i++)
            {
                var boat = prichalStages[currentLevel][i];
                if (boat != null)
                {
                    boat.setPosition(5 + i / 5 * placeSizeWidth + 5, i % 5 * placeSizeHeight + 15);
                    boat.drawBoat(g);
                }
            }
        }
        private void DrawMarking(Graphics g)
        {
            Pen pen = new Pen(Color.Black, 3);
            g.DrawString("L" + (currentLevel + 1), new Font("Arial", 30), new SolidBrush(Color.Blue),
                (countPlaces / 5) * placeSizeWidth - 70, 420);
            g.DrawRectangle(pen, 0, 0, (countPlaces / 5) * placeSizeWidth, 480);
            for (int i = 0; i < countPlaces / 5; i++)
            {
                for (int j = 0; j < 6; j++)
                {
                    g.DrawLine(pen, i * placeSizeWidth, j * placeSizeHeight, i * placeSizeWidth + 110, j * placeSizeHeight);
                    if (j < 5)
                    {
                        g.DrawString((i * 5 + j + 1).ToString(), new Font("Arial", 30), new SolidBrush(Color.Blue),
              i * placeSizeWidth + 30, j * placeSizeHeight + 20);
                    }

                }
                g.DrawLine(pen, i * placeSizeWidth, 0, i * placeSizeWidth, 400);
            }
        }
    }
}

File: TP_laba2/src/Prichal.java
import java.awt.Color;
import java.awt.Graphics;

public class Prichal {
	ClassArray<Transport> prichal;
	int countPlaces = 20;

    int placeSizeWidth = 210;
    int placeSizeHeight = 80;
	public Prichal() {
		prichal = new ClassArray<Transport>(countPlaces, null);
	}

	public int PutNewBoatInPrichal(Transport boat) {
		return prichal.add(prichal, boat);
	}

	public Transport GetNewBoatFromPrichal(int index) {
		return prichal.dec(prichal, index);
	}

	public void Draw(Graphics g) {
		drawPrichal(g);
		for (int i = 0; i < countPlaces; i++) {
			Transport boat = prichal.getObject(i);
			if (boat != null) {
				boat.setPosition(5 + i / 5 * placeSizeWidth + 15, i % 5 * placeSizeHeight + 30);
				boat.drawVehicle(g);
			}
		}
	}

	public void drawPrichal(Graphics g) {
	       
        g.setColor(Color.WHITE);

        g.fillRect(0, 0, (countPlaces / 5) * placeSizeWidth, 480);
        g.setColor(Color.black);
        for (int i=0;i<countPlaces/5;i++) {
            for (int j=0;j<6;++j) {
                g.drawLine(i * placeSizeWidth, j * placeSizeHeight, i * placeSizeWidth + 110, j * placeSizeHeight);
            }
            g.drawLine(i*placeSizeWidth,0,i*placeSizeWidth,400);
        
    }
	}
}
